base/test/atomic_wait_benchmark.cc

```diff
@@ -0,0 +1,104 @@
+#include <benchmark/benchmark.h>
+
+#include <atomic>
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+#include "base/atomic_wait.h"
+
+namespace {
+
+class AtomicPingPong {
+  static constexpr int Ping = 0;
+  static constexpr int Pong = 1;
+
+  alignas(64) std::atomic<int> turn_{Ping};
+  alignas(64) std::atomic<int> rounds_left_;
+
+public:
+  explicit AtomicPingPong(int rounds) : rounds_left_(rounds) {}
+
+  void ping() {
+    while (rounds_left_ > 0) {
+      slinky::atomic_wait(&turn_, Pong);
+      --rounds_left_;
+      turn_.store(Pong);
+      slinky::atomic_notify_one(&turn_);
+    }
+  }
+
+  void pong() {
+    while (rounds_left_ > 0) {
+      slinky::atomic_wait(&turn_, Ping);
+      turn_.store(Ping);
+      slinky::atomic_notify_one(&turn_);
+    }
+  }
+};
+
+class CondVarPingPong {
+  static constexpr int Ping = 0;
+  static constexpr int Pong = 1;
+
+  std::mutex m_;
+  std::condition_variable cv_;
+  int turn_{Ping};
+
+  alignas(64) std::atomic<int> rounds_left_;
+
+public:
+  explicit CondVarPingPong(int rounds) : rounds_left_(rounds) {}
+
+  void ping() {
+    while (rounds_left_ > 0) {
+
+      std::unique_lock<std::mutex> lock(m_);
+      cv_.wait(lock, [&] { return turn_ == Ping; });
+
+      --rounds_left_;
+      turn_ = Pong;
+
+      lock.unlock();
+      cv_.notify_one();
+    }
+  }
+
+  void pong() {
+    while (rounds_left_ > 0) {
+
+      std::unique_lock<std::mutex> lock(m_);
+      cv_.wait(lock, [&] { return turn_ == Pong; });
+
+      --rounds_left_;
+      turn_ = Ping;
+
+      lock.unlock();
+      cv_.notify_one();
+    }
+  }
+};
+
+template <class PingPong> void BM_ping_pong(benchmark::State &state) {
+  const int rounds = state.range(0);
```

Comment:
I think this is probably better written like this benchmark: https://github.com/dsharlet/slinky/blob/b8bf32c049dd03c533d07342a3295e91ebc113ec/base/test/arithmetic_benchmark.cc#L11-L15

So you don't need to specify the `rounds`/`Iterations`.